src/server/scripts/EasternKingdoms/SunwellPlateau/boss_muru.cpp

```diff
@@ -440,6 +321,29 @@ class spell_entropius_black_hole_effect : public SpellScript
     }
 };
 
+// 46284 - Negative Energy Periodic
+class spell_entropius_negative_energy_periodic : public AuraScript
+{
+    PrepareAuraScript(spell_entropius_negative_energy_periodic);
+
+    bool Validate(SpellInfo const* spellInfo) override
+    {
+        return ValidateSpellInfo({ spellInfo->Effects[EFFECT_0].TriggerSpell });
+    }
+
+    void PeriodicTick(AuraEffect const* aurEff)
+    {
+        PreventDefaultAction();
+        uint32 targetCount = aurEff->GetTickNumber() > 12 ? 1 : aurEff->GetTickNumber() / 12;
+        GetTarget()->CastCustomSpell(aurEff->GetSpellInfo()->Effects[EFFECT_0].TriggerSpell, SPELLVALUE_MAX_TARGETS, targetCount);
```

Comment:
Is null check for target required here?

Reply:
No because the aura will never tick if the target it is applied to doesn't exist, it is an illogical situation